test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -245,5 +276,267 @@ g.test_vote_during_txn_wal_write = function(g)
+    --
+    -- Server1 doesn't see the new term yet and makes an attempt to do a sync
+    -- transaction.
+    --
+    local f_insert = fiber.new(g.server1.exec, g.server1, function()
+        box.space.test:replace{1}
+    end)
+    f_insert:set_joinable(true)
+    --
+    -- Restore connectivity server1 -> server2.
+    --
+    server_set_replication(g.server2, g.server2.box_cfg.replication)
+    --
+    -- Server2 gets the bad txn.
+    --
+    g.server1:wait_downstream_to(g.server2)
+    t.assert_equals(g.server2:exec(function()
+        return box.space.test:count()
+    end), 1)
+    --
+    -- PROMOTE finally is written. The server1's transaction is rolled back
+    -- gracefully and no split brain has happened.
+    --
+    server_unblock_wal_write(g.server3)
+    local ok, err = f_promote:join()
+    t.assert_equals(err, nil)
+    t.assert(ok)
+    g.server3:wait_election_leader()
+    --
+    -- Server1 gets the PROMOTE and fails its rogue txn.
+    --
+    server_set_replication(g.server1, g.server1.box_cfg.replication)
+    ok, err = f_insert:join()
+    t.assert_not_equals(err, nil)
+    t.assert(not ok)
+    t.assert_equals(g.server1:exec(function()
+        return box.space.test:count()
+    end), 0)
+    --
+    -- Cleanup.
+    --
+    server_set_replication(g.server3, g.server3.box_cfg.replication)
+    g.server3:exec(function()
+        box.cfg{election_mode = 'voter'}
+    end)
+    g.server1:exec(function()
+        box.ctl.promote()
+    end)
+    g.server1:wait_election_leader()
+    wait_fullmesh(g)
+end
+
+--
+-- Similar to the test case without '_complex' suffix. But the transaction is
+-- delivered to a conflicting node in an intricate way. Server1 - old leader,
+-- server3 - new leader, server2 - between them.
+--
+-- In this test server2 gets the bad txn not directly from server1, but via
+-- another instance (server4). Server4 gets the txn from server1, then gets
+-- its term bumped by server2, then sends the txn to server2. This way server2
+-- sees that the txn is coming from an instance having the new term.
+--
+-- The purpose is to ensure that server2 anyway later will send ACK with a new
+-- term to server1. It wouldn't be enough to simply reject any txns from nodes
+-- having an old term, because in this test server4 has a new term but still
+-- forwards the bad txn.
+--
+g.test_old_leader_txn_during_promote_write_complex = function(g)
+    local server1_uri = server.build_instance_uri('server1')
+    local server2_uri = server.build_instance_uri('server2')
+    local server3_uri = server.build_instance_uri('server3')
+    local server4_uri = server.build_instance_uri('server4')
+    local server5_uri = server.build_instance_uri('server5')
+    --
+    -- Build the topology:
+    --   server4 <- fullmesh(server1, server2, server3)
+    --   server3 <-> server5
+    --
+    g.server4 = g.cluster:build_and_add_server({
+        alias = 'server4', box_cfg = table.deepcopy(g.server3.box_cfg)
```

Comment:
Deepcopy is done by `cluster:build_server` anyway, you don't have to repeat it here

Reply:
Indeed, dropped the `deepcopy`.